### PR TITLE
OpenShift support for Spire

### PR DESCRIPTION
**Pull request by sabre1041**
**Pull Request check list**

- [ x ] Commit conforms to CONTRIBUTING.md?
- [ ] Proper tests/regressions included?
- [ x ] Documentation updated?

**Affected functionality**
Use of Spire on OpenShift

**Description of change**
Documenting policies for enabling Spire on OpenShift

**Which issue this PR fixes**
N/A

### CODE REVIEW

File: doc/spire_agent.md [hunk 1]
```diff
@@ -272,6 +272,108 @@ resource name (e.g. `spiffe://example.org`). Alternatively, if the default name
...
+  - kind: ServiceAccount
+    name: spire-agent
+    namespace: spire
+```
+
+As SCC's are applied at pod admission time, remove any existing Spire Agent pods. All newly admitted pods will make use of the `spire` SCC enabling their use within OpenShift.
+
 ## Further reading
 
 * [SPIFFE Reference Implementation Architecture](https://docs.google.com/document/d/1nV8ZbYEATycdFhgjTB619pwIvamzOjU6l0SyBGbzbo4/edit#)
```
Comment (dfeldman) [inline doc/spire_agent.md:379]: any chance this doc could be made public?

Comment (sabre1041) [reply doc/spire_agent.md:379]: @dfeldman Which document are you referring to? The "SPIFFE Reference Implementation Architecture" reference is not part of this PR

**Comment by evan2645** [comment]
Thank you very much for this contribution @sabre1041! I'm currently trying to figure out where we should be keeping this, if it's discoverable here etc...

**Comment by evan2645** [comment]
I think this should go on the k8s guide of the spiffe.io website. We'll take it here for now though so it's captured somewhere, and see about getting it moved over.

Thank you for this @sabre1041!

**Review by evan2645 — APPROVED** [review]

**Pull request merged**